stackrox/stackrox
# ROX-11986: Fix minimum access role updates

## Description

When trying to update the minimum access role within the UI, the updated value of the minimum access role is not actually reflected within the request:
![Screenshot 2022-07-27 at 02 03 32](https://user-images.githubusercontent.com/89904305/181133748-9bea29da-434f-4814-b108-6c49941f8ee4.png)

Instead, the same value is being sent.

Additionally, when trying to update an existing default group via the `/v1/groupsbatch` API, the update will fail with the following error message:
```json
{
    "error": "a default group already exists for auth provider \"3de3432f-5e5b-4903-bbde-82bcc232b239\"",
    "code": 6,
    "message": "a default group already exists for auth provider \"3de3432f-5e5b-4903-bbde-82bcc232b239\"",
    "details": []
}
```

This is due to the fact that we only check for the existence of a default group in the update case. Instead, we should allow updates to the existing group.

This PR will introduce the following:
- Fixing the role name sent in the `groupsbatch` request from the UI.
- Adding E2E tests for changing the minimum access role.
- Allow updates to existing default groups.

## Checklist

If any of these don't apply, please comment below.

## Testing Performed
- see Unit tests.
- E2E test for UI changes.
- Manual tests:
1. Create an auth provider of your choice.
2. Update the minimum access role.
3. Ensure the minimum access role is correctly set within UI + API requests.
